app/scripts/controllers/deploymentConfig.js

```diff
@@ -63,6 +63,17 @@ angular.module('openshiftConsole')
     });
     AlertMessageService.clearAlerts();
 
+    // copy deploymentConfig and ensure it has env so that we can edit env vars using key-value-editor
+    var copyDeploymentConfigAndEnsureEnv = function(deploymentConfig) {
+      $scope.updatedDeploymentConfig = angular.copy(deploymentConfig);
+      $scope.updatedDeploymentConfig.spec.template.spec.containers = _.map($scope.updatedDeploymentConfig.spec.template.spec.containers, function(container){
```

Comment:
wouldn't this be easier to read as
```
_.each($scope.updatedDeploymentConfig.spec.template.spec.containers, function(container) {
  container.env = container.env || [];
}
```
updatedDeploymentConfig is already a copy, i don't see the need for creating a new array (which i assume was the reason for using map in the first place)?  @benjaminapetersen  ?